src/read-only-handler.ts

```diff
@@ -42,7 +42,11 @@ export class ReadOnlyHandler {
             const { originalTarget } = this;
             throw new Error(`Invalid mutation: Cannot set "${key.toString()}" on "${originalTarget}". "${originalTarget}" is read-only.`);
         }
-        return false;
+
+        // JSC doesn't throw a type Error when the set handler returns false. Let's throw in the handler to be
+        // consistent with the rest of the engines.
+        // https://bugs.webkit.org/show_bug.cgi?id=177398
+        throw new TypeError('Invalid mutation on readonly proxy.');
```

Comment:
this is also not the same error that they will see in other browsers today.

Reply:
I think we should be fine here. The [set operation](https://tc39.github.io/ecma262/#sec-set-o-p-v-throw) defines that it should throw a `TypeError` is `success` is `false`, the error message is up to the implementer.